### DISCUSSION TITLE
Transient prompt error with Constrained Language Mode since version 9.3.1

### DISCUSSION THREAD

**Discussion by SebCT** [JanDeDobbeleer/oh-my-posh#2801] [Troubleshoot] [answered]
Hello!

Since the last update to version 9.3.1 i saw errors with themes that use the transient prompt - in Line 125 and 143 in the omp.ps1:

<img width="1350" alt="image" src="https://user-images.githubusercontent.com/62695489/190834167-476962c2-a22f-4e45-a64c-3c2ac8ce950b.png">

Themes without that transient prompt are working without an issue. in Full language Mode the transient prompt is also working fine, as i can remember in Version 9.3.0 worked at least the 1_shell theme.

**Comment by lewis-yeung** [comment]
@JanDeDobbeleer Should we disable the transient prompt for `ConstrainedLanguage` mode? Moreover, should we handle those cases for other [language modes in PowerShell](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_language_modes?view=powershell-7.2#language-modes)?

**Reply by JanDeDobbeleer** [reply]
Disabling is a good idea, I would not look at other modes as this is (today) the only relevant one.

**Reply by SebCT** [reply]
Maybe in one of the future releases it could be possible, that the ps1 script runs signed, so that it could be allowlisted and automatically in FullLanguage Mode? But i have no ideo how to solve this because of the EXE wrapper - i always code sign my PS1 scripts for FullLanguage Mode because of AppLocker/WDAC.

**Comment by lewis-yeung** [comment, marked as answer]
@SebCT Fixed in [v9.3.2](https://github.com/JanDeDobbeleer/oh-my-posh/releases/tag/v9.3.2).

**Reply by JanDeDobbeleer** [reply]
@lewis-yeung I was thinking, does this imply the same for tooltips?

**Reply by lewis-yeung** [reply]
@JanDeDobbeleer I've just tested it and yes, it does. We have to fix this, too.

**Reply by lewis-yeung** [reply]
Also, we should add some clarification to the docs.

**Reply by SebCT** [reply]
Thank you!
